--- makefile.in.orig	2010-08-04 06:20:44.000000000 -0600
+++ makefile.in	2011-01-27 10:16:18.353728096 -0700
@@ -8,10 +8,10 @@
 	rm -f makefile config.h
 	rm -f gmon.out *~ 
 
-picosat: libpicosat.a app.o main.o
+picosat: libpicosat.so app.o main.o
 	$(CC) $(CFLAGS) -o $@ main.o app.o -L. -lpicosat
 
-picomus: libpicosat.a picomus.o
+picomus: libpicosat.so picomus.o
 	$(CC) $(CFLAGS) -o $@ picomus.o -L. -lpicosat
 
 app.o: app.c picosat.h makefile
@@ -24,10 +24,10 @@
 	$(CC) $(CFLAGS) -c $<
 
 picosat.o: picosat.c picosat.h makefile
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS) -fPIC -c $<
 
 version.o: version.c config.h makefile
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS) -fPIC -c $<
 
 config.h: makefile VERSION mkconfig # and actually picosat.c
 	rm -f $@; ./mkconfig > $@
@@ -38,6 +38,6 @@
 
 SONAME=-Xlinker -soname -Xlinker libpicosat.so
 libpicosat.so: picosat.o version.o
-	$(CC) $(CFLAGS) -shared -o $@ picosat.o version.o $(SONAME)
+	$(CC) $(CFLAGS) -fPIC -shared -o $@ picosat.o version.o $(SONAME)
 
 .PHONY: all clean
--- picosat.c.orig	2010-08-04 06:20:44.000000000 -0600
+++ picosat.c	2011-01-27 10:01:38.659957334 -0700
@@ -6481,7 +6481,7 @@
   (void) file;
   (void) fmt;
   (void) f;
-  ABORT ("compiled without trace support");
+  ABORT ("compiled without trace support; please use picosat.trace instead");
 #endif
 }
 
@@ -6900,7 +6900,7 @@
       leave ();
   }
 #else
-  ABORT ("compiled without trace support");
+  ABORT ("compiled without trace support; please use picosat.trace instead");
 #endif
 
   return res;
@@ -6938,7 +6938,7 @@
       leave ();
   }
 #else
-  ABORT ("compiled without trace support");
+  ABORT ("compiled without trace support; please use picosat.trace instead");
 #endif
 
   return res;
