--- makefile.in.orig
+++ makefile.in
@@ -12,13 +12,13 @@
 analyze:
 	clang --analyze $(CFLAGS) *.c *.h
 
-picosat: libpicosat.a app.o main.o
+picosat: libpicosat.so app.o main.o
 	$(CC) $(CFLAGS) -o $@ main.o app.o -L. -lpicosat
 
 picomcs: libpicosat.a picomcs.o
 	$(CC) $(CFLAGS) -o $@ picomcs.o -L. -lpicosat
 
-picomus: libpicosat.a picomus.o
+picomus: libpicosat.so picomus.o
 	$(CC) $(CFLAGS) -o $@ picomus.o -L. -lpicosat
 
 picogcnf: libpicosat.a picogcnf.o
@@ -40,10 +40,10 @@
 	$(CC) $(CFLAGS) -c $<
 
 picosat.o: picosat.c picosat.h makefile
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS) -fPIC -c $<
 
 version.o: version.c config.h makefile
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS) -fPIC -c $<
 
 config.h: makefile VERSION mkconfig.sh # and actually picosat.c
 	rm -f $@; ./mkconfig.sh > $@
@@ -54,6 +54,6 @@
 
 SONAME=-Xlinker -soname -Xlinker libpicosat.so
 libpicosat.so: picosat.o version.o
-	$(CC) $(CFLAGS) -shared -o $@ picosat.o version.o $(SONAME)
+	$(CC) $(CFLAGS) -fPIC -shared -o $@ picosat.o version.o $(SONAME)
 
 .PHONY: all clean
--- picosat.c.orig
+++ picosat.c
@@ -6547,7 +6547,7 @@
   (void) file;
   (void) fmt;
   (void) f;
-  ABORT ("compiled without trace support");
+  ABORT ("compiled without trace support; please use picosat.trace instead");
 #endif
 }
 
@@ -7262,7 +7262,7 @@
     return res;
   }
 #else
-  ABORT ("compiled without trace support");
+  ABORT ("compiled without trace support; please use picosat.trace instead");
   return 0;
 #endif
 }
@@ -7298,7 +7298,7 @@
     return res;
   }
 #else
-  ABORT ("compiled without trace support");
+  ABORT ("compiled without trace support; please use picosat.trace instead");
   return 0;
 #endif
 }
